teps/0120-canceling-concurrent-pipelineruns.md

```diff
@@ -102,7 +102,307 @@ queueing, or more advanced concurrency strategies.
 
 ## Proposal
 
-TODO
+Create an experimental project (either a new Github repo or a new subfolder of the experimental repo) with its own reconciler for canceling concurrent PipelineRuns.
+The strategy used in this experimental project will be most similar to the ["Separate concurrency CRD and reconciler" strategy](#separate-concurrency-crd-and-reconciler) below.
+Use this project on our dogfooding cluster for CI PipelineRuns, and allow this experience to inform how we want to make this part of our API.
+
+[TEP-0098: Workflows](./0098-workflows.md) proposes creating a Workflows API for an easier end-to-end, getting started experience with Tekton.
+We will likely want to allow users to configure concurrency controls on Workflows.
+This work should wait until the Workflows API has been implemented and until we have a better sense of experimental results for the initial project on our dogfooding cluster.
+
+Before marking this TEP as implementable, we should answer the following questions:
+- Do we want to allow a PipelineRun to be part of multiple concurrency groups?
+  - For example, when we later support queueing, we might want to allow users to configure a rule like "Cancel all but the last PR per pull request, and only allow 5 CI runs per repo at a time."
+- What concurrency strategies should we support?
+  - For the initial version of this proposal, we will only support cancelation. However, should we support all possible strategies for canceling PipelineRuns,
+    including cancelation, graceful cancelation, and graceful stopping?
+- Should we attempt to prevent users from interfering with concurrency controls?
+  - Many of the proposed solutions rely on labels, and a user editing labels could change PipelineRun behavior. Is this desired?
```

Comment:
I think so yes. A user can change labels on a `Pod` from a `TaskRun` and provoke a weird behavior for example. This is very similar here, we cannot completely stop the user from shooting its foot, and that's fine 🙃